### PR TITLE
Add APIs to handle PROCESSOR_NUMBER

### PR DESCRIPTION
**Pull request by Alan-Jowett**
Add mocks for:
KeGetProcessorNumberFromIndex
KeGetProcessorIndexFromNumber
KeSetTargetProcessorDpcEx

Assumes group == 64 processors.

### CODE REVIEW

File: src/ke.cpp [hunk 1]
```diff
@@ -620,6 +620,25 @@ class _usersim_emulated_dpc
...
+    ProcNumber->Group = (WORD)(ProcessorIndex / 64);
+    ProcNumber->Number = (UCHAR)(ProcessorIndex % 64);
+    ProcNumber->Reserved = 0;
+    return STATUS_SUCCESS;
+}
+
+ULONG
+KeGetProcessorIndexFromNumber(_In_ PPROCESSOR_NUMBER ProcNumber)
+{
+    return (ULONG)ProcNumber->Group * 64 + (ULONG)ProcNumber->Number;
```
Comment (nibanks) [inline src/ke.cpp:639]: It's unfortunately much more complicated than this. Groups may have 40 processors for instance, in an 80 processor machine.

Comment (nibanks) [reply src/ke.cpp:639]: See msquic code here: https://github.com/microsoft/msquic/blob/main/src/platform/platform_winuser.c#L81

Comment (Alan-Jowett) [reply src/ke.cpp]: Should be fixed now.

File: src/platform_user.cpp [hunk 2]
```diff
@@ -381,7 +381,39 @@ KeGetCurrentProcessorNumberEx(_Out_opt_ PPROCESSOR_NUMBER ProcNumber)
...
+{
+    // Compute the CPU index from the group and number.
+    return _usersim_platform_group_to_index_map[ProcNumber->Group] + ProcNumber->Number;
+}
+
+NTSTATUS
+KeGetProcessorNumberFromIndex(ULONG ProcessorIndex, _Out_ PPROCESSOR_NUMBER ProcNumber)
+{
+
+    if (ProcessorIndex >= _usersim_platform_maximum_processor_count) {
```
Comment (nibanks) [inline src/platform_user.cpp:398]:
```suggestion
{
    if (ProcessorIndex >= _usersim_platform_maximum_processor_count) {
```

**Review by nibanks — APPROVED** [review]

**Pull request merged**